view/frontend/web/js/model/shipping-save-processor/default.js

```diff
@@ -25,21 +25,32 @@ define(
         'ClassyLlama_AvaTax/js/view/checkout-validation-handler',
         'Magento_Ui/js/modal/alert'
     ],
-    function (
-        $,
-        quote,
-        resourceUrlManager,
-        storage,
-        paymentService,
-        methodConverter,
-        errorProcessor,
-        fullScreenLoader,
-        selectBillingAddressAction,
-        checkoutValidationHandler,
-        alert
-    ) {
+    function ($,
+              quote,
+              resourceUrlManager,
+              storage,
+              paymentService,
+              methodConverter,
+              errorProcessor,
+              fullScreenLoader,
+              selectBillingAddressAction,
+              checkoutValidationHandler,
+              alert) {
         'use strict';
 
+        var getPayloadExtender = function() {
+            var extenderRequest = new $.Deferred();
+            require(
+                ['Magento_Checkout/js/model/shipping-save-processor/payload-extender'],
```

Comment:
@sjb9774 You may be able to use 
`require.specified('Magento_Checkout/js/model/shipping-save-processor/payload-extender')`
to check if that module exists or not. That should return true/false and would prevent that 404 error on the console. I tested it on 2.2 and it returned what I thought it should, but I don't have a 2.1 instance set up right now. Could you test it and update the PR if it is useful?

Reply:
@prestonchoate doesn't require.specified only determine if it has already been loaded? If it hasn't been loaded yet, but is available to be loaded, this would not run?

Reply:
A 404 seems unavoidable, as the only possible way for the frontend to absolutely know if something exists is to attempt to load it? Unless Magento somehow scrapes the entire directories to build a listing of every possible file to load? Which would seem like a bad practice...

Reply:
@LordZardeck it appears you are correct. I thought that require.defined() only returned true when it was loaded and .specified() would return true when it was an available file.